R/Design-methods.R

```diff
@@ -18,37 +18,40 @@ NULL
 #' This code is basically copied from `stats:::simulate.lm`.
 #'
 #' @param seed an object specifying if and how the random number generator
-#' should be initialized (\dQuote{seeded}). Either `NULL` (default) or an
-#' integer that will be used in a call to [`set.seed`] before
+#' should be initialized ("seeded"). Either `NULL` (default) or an
+#' integer that will be used in a call to [set.seed()] before
 #' simulating the response vectors. If set, the value is saved as the
 #' `seed` slot of the returned object. The default, `NULL` will
 #' not change the random generator state.
-#' @return The RNGstate will be returned, in order to call this function
-#' with this input to reproduce the obtained simulation results.
+#' @return The integer vector containing the random number generate state will
+#' be returned, in order to call this function with this input to reproduce
+#' the obtained simulation results.
 #'
 #' @export
 #' @keywords programming
 set_seed <- function(seed = NULL) {
+  assert_number(seed, null.ok = TRUE)
+
   if (!exists(".Random.seed", envir = .GlobalEnv, inherits = FALSE)) {
     runif(1)
   }
 
-  if (is.null(seed)) {
-    rng_state <- get(".Random.seed", envir = .GlobalEnv)
+  rng_state <- if (is.null(seed)) {
+    get(".Random.seed", envir = .GlobalEnv)
   } else {
+    seed <- as.integer(seed)
     r_seed <- get(".Random.seed", envir = .GlobalEnv)
     # Make sure r_seed exists in parent frame.
-    assign("r_seed", r_seed, envir = parent.frame())
+    assign(".r_seed", r_seed, envir = parent.frame())
     set.seed(seed)
-    rng_state <- structure(seed, kind = as.list(RNGkind()))
-    do.call("on.exit",
-      list(quote(assign(".Random.seed", r_seed, envir = .GlobalEnv))),
+    # Here we need the r_seed in the parent.frame!
+    do.call(
+      "on.exit",
+      list(quote(assign(".Random.seed", .r_seed, envir = .GlobalEnv))),
       envir = parent.frame()
     )
-    # Here we need the r_seed in the parent.frame!
+    structure(seed, kind = as.list(RNGkind()))
   }
-
-  return(rng_state)
```

Comment:
we should still return this, just without explicit `return` statement, so
just `rng_state` here is fine. Alternatively can also omit the `rng_state` variable assignment in line 39 altogether, then the last value is the thing from the if/else outcome, so what we want

Reply:
and I guess it is actually returned implicitly right now, but it is not easily readable as that...

Reply:
I removed `rng_state` variable assignment.